RealmGridViewExample: Demo crash by java.lang.IllegalStateException.

When I click the item on GridView and exit the demo app, if you open app again, below crash error throw:
**
     Caused by: java.lang.IllegalStateException: It's not allowed to delete the file associated with an open Realm. Remember to close() all the instances of the Realm before deleting its file.
            at io.realm.Realm.deleteRealm(Realm.java:1722)
            at io.realm.examples.realmgridview.GridViewExampleActivity.onCreate(GridViewExampleActivity.java:60)
            at android.app.Activity.performCreate(Activity.java:5283)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1088)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2307)
**

And when item click, below code execute:
```java
// Update the realm object affected by the user
Realm realm = Realm.getInstance(this);

// Acquire the list of realm cities matching the name of the clicked City.
City city = realm.where(City.class).equalTo("name", modifiedCity.getName()).findFirst();

// Create a transaction to increment the vote count for the selected City in the realm
realm.beginTransaction();
city.setVotes(city.getVotes() + 1);
realm.commitTransaction();
```
Do we need to call close() when done using the Realm instance?

Realm version: 0.82.1

Any particular reason you think it is low-level? It describes what is wrong, but any suggestions for improving it would be welcome. 

```
Caused by: java.lang.IllegalStateException: It's not allowed to delete the file associated with an open Realm. Remember to close() all the instances of the Realm before deleting its file.
```

--- a/examples/gridViewExample/src/main/java/io/realm/examples/realmgridview/GridViewExampleActivity.java
+++ b/examples/gridViewExample/src/main/java/io/realm/examples/realmgridview/GridViewExampleActivity.java
@@ -131,8 +131,6 @@ public class GridViewExampleActivity extends Activity implements AdapterView.OnI
     }
 
     public void updateCities() {
-        Realm realm = Realm.getInstance(this);
-
         // Pull all the cities from the realm
         RealmResults<City> cities = realm.where(City.class).findAll();
 
@@ -145,9 +143,6 @@ public class GridViewExampleActivity extends Activity implements AdapterView.OnI
     @Override
     public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
         City modifiedCity = (City)mAdapter.getItem(position);
-
-        // Update the realm object affected by the user
-        Realm realm = Realm.getInstance(this);
 
         // Acquire the list of realm cities matching the name of the clicked City.
         City city = realm.where(City.class).equalTo("name", modifiedCity.getName()).findFirst();

